portlayer/storage/image.go

```diff
@@ -34,6 +34,9 @@ type Image struct {
 	Parent *url.URL
 
 	Store *url.URL
+
+	// Metadata associated with the image.
+	Metadata map[string][]byte
```

Comment:
The key/vals are immutable per disk image we create.  So use whatever key you want so long as it's the same key you use to retrieve the blob.   The storage layer will persist whatever key/blob you use in writeimage and return it in the `Image` struct.  So, as long as you are consistent, it shouldn't matter.  

It's likely we'll need more than just 1 blob per image.  So `docker_tarfiles` for the docker metadata and `docker_overrides` for things like flags passed on the cli would be good examples.